homeassistant/helpers/config_validation.py

```diff
@@ -385,8 +387,47 @@ def validator(value):
     return validator
 
 
+def list_or_comma_separated(member_validator=match_all) -> Callable:
+    """Return a validator.
+
+    The validator checks that the value is a list or a comma-separated string
+    that fits member_validator.
+    """
+    def validate(value: Union[str, Sequence]) -> Sequence:
+        """Validate string or sequence."""
+        if value is None:
+            raise vol.Invalid('Value can not be None')
+        if isinstance(value, str):
+            value = [member.strip() for member in value.split(',')]
+        try:
+            return [member_validator(member) for member in value]
+        except (ValueError, TypeError):
+            raise vol.Invalid(
+                'Expected sequence or string, got {}'.format(value))
+
+    return validate
+
+
+def validate_some_keys(keys: Sequence[str], validator: Callable) -> Callable:
```

Comment:
This is very close to `ordered_dict` and suffers the same problem as groups... voluptuous will kill the order. So this is needed.

But I suggest we add a parameter to `ordered_dict` to support this case, say `key_list`?

```python
-def ordered_dict(value_validator, key_validator=match_all):
+def ordered_dict(value_validator, key_validator=match_all, key_list=None):
+         # key_list provides an optional list of keys to which validation will be restricted
         ...
         for key, val in value.items():
-             v_res = item_validator({key: val})
+             skip_key_validation = key_list and key not in key_list
+             v_res = {key:val} is skip_key_validation else item_validator({key: val})
             config.update(v_res)
```